### DISCUSSION TITLE
How to limit result set?

### DISCUSSION THREAD

**Discussion by Kolos65** [msiemens/tinydb#384] [Q&A] [answered]
Hi!
Can I limit the number of items to return from a query?
For example: db.all(limit=10) would return the 10 newest records in the db.
Awesome project btw!👏

**Comment by Kolos65** [comment]
Or does it load everything in memory so this doesnt make sense?

**Comment by therden** [comment, marked as answer]
Assuming that you're using the default `JSONStorage` and that you haven't set `sort_keys=True,` then (since `db.all()` returns a list of dicts), you can slice the results:

db.all()[-10:]   # will return a list of the ten most recently inserted documents 
db.all()[-3:]   # will return a list of the three most recently inserted documents

**Reply by Kolos65** [reply]
Thanks for the answer! I was thinking there might be a limit parameter which sets the number of items to read from the json file, so if there are lots of stored items, it only reads the amount needed.

**Reply by therden** [reply]
You're welcome.  

As per [this reply from TinyDB's author](https://github.com/msiemens/tinydb/discussions/376#discussioncomment-419172) to another question, the entire file is read on each read operation.

**Comment by therden** [comment]
Query results would work the same way:

`db.search(Query().priority == "1")[-3:]`

will display a list of the three most recently inserted documents among the query results.
